src/internal/memory/memory_block.hpp

```diff
@@ -19,13 +19,13 @@
 
 #include <stdexcept>
 
-namespace srf::memory {
+namespace srf::internal::memory {
 
-struct memory_block
+struct MemoryBlock
```

Comment:
- Partially covered by some tests, but nothing that covers `distance` or `offset` methods.

Reply:
Will resolve with #109